### DISCUSSION TITLE
Avoid invalid mapping types

### DISCUSSION THREAD

**Discussion by PaulVrugt** [riok/mapperly#659] [Q&A]
Currently, when I have a mapping like this:

```c#
[Mapper]
public partial class Mapper
{
    public partial InvalidTarget ToTarget(TestSource source);
}

public class TestSource
{
    public Guid Id { get; set; }
}

public class InvalidTarget
{
    public RandomClass Id { get; set; }
}

public class RandomClass { }
```

mapperly generates:

```c#
public partial class Mapper
    {
        public partial global::MapperlyTest.InvalidTarget ToTarget(global::MapperlyTest.TestSource source)
        {
            var target = new global::MapperlyTest.InvalidTarget();
            target.Id = MapToRandomClass(source.Id);
            return target;
        }

        private global::MapperlyTest.RandomClass MapToRandomClass(global::System.Guid source)
        {
            var target = new global::MapperlyTest.RandomClass();
            return target;
        }
    }
```

It maps the `Id` property without errors, but just creates a new `RandomClass` instance, and doesn't do anything with the source value. I understand it cannot do anything with it, but it also doesn't throw an error. I would like to have some way to avoid this seeming to be a valid mapping. The conversions documentation states you can avoid some automatic conversions, but I don't seem to be able to disable the `New instance` conversion. Also, the the `New instance` conversion also seems to be the one that is responsible for checking for a factory for the class. I would NOT want to disable this.

**Comment by latonz** [comment]
Created https://github.com/riok/mapperly/issues/660 to emit a diagnostic if an object mapping is created without mapping any members.
